test/e2e_validate_data/setup_testcontainers.py

```diff
@@ -0,0 +1,205 @@
+import subprocess
+from contextlib import contextmanager
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterator
+
+import docker
+import regex
+import requests
+from docker.errors import NotFound
+from docker.models.networks import Network
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.waiting_utils import wait_for_logs
+
+E2E_TESTDATA = Path("testdata/e2e").absolute()
+SIPI_IMAGES = E2E_TESTDATA / "images"
+TMP_SIPI = E2E_TESTDATA / "tmp-dsp-sipi"
+TMP_INGEST = E2E_TESTDATA / "tmp-dsp-ingest"
+INGEST_DB = E2E_TESTDATA / "ingest-db"
+
+
+@dataclass(frozen=True)
+class ImageVersions:
+    fuseki: str
+    sipi: str
+    ingest: str
+    api: str
+
+
+@dataclass(frozen=True)
+class Containers:
+    fuseki: DockerContainer
+    sipi: DockerContainer
+    ingest: DockerContainer
+    api: DockerContainer
+
+
+@contextmanager
+def get_test_network() -> Iterator[Network]:
+    name = "dsp-tools-test-network"
+    client = docker.client.from_env()
+    try:
+        network = client.networks.get(name)
+    except NotFound:
+        client.networks.create(name, internal=True, check_duplicate=True)
+        network = client.networks.get(name)
+    try:
+        yield network
+    finally:
+        network.remove()
+        client.close()  # type: ignore[no-untyped-call]  # incomplete stubs - remove this when the stubs are complete
+
+
+@contextmanager
+def get_containers() -> Iterator[Containers]:
+    if subprocess.run("docker stats --no-stream".split(), check=False).returncode != 0:
+        raise RuntimeError("Docker is not running properly")
+    with get_test_network() as network:
+        containers = _get_all_containers(network)
+        try:
+            yield containers
+        finally:
+            _stop_all_containers(containers)
+
+
+def _get_all_containers(network: Network) -> Containers:
+    versions = _get_image_versions()
+    fuseki = _get_fuseki_container(network, versions.fuseki)
+    sipi = _get_sipi_container(network, versions.sipi)
+    ingest = _get_ingest_container(network, versions.ingest)
+    api = _get_api_container(network, versions.api)
+    containers = Containers(fuseki=fuseki, sipi=sipi, ingest=ingest, api=api)
+    _print_containers_are_ready(containers)
+    return containers
+
+
+def _get_image_version(docker_compose_content: str, component: str) -> str:
+    match = regex.search(rf"image: daschswiss/{component}:([^\n]+)", docker_compose_content)
+    return match.group(1) if match else "latest"
+
+
+def _get_api_image_version() -> str:
```

Comment:
This is the only difference to the other set-up

Reply:
*question*: Would you be able to remove this by using the latest dsp-api release?

This is a short term solution and should be gone in a couple of days, so it is fine to me as long as we are removing it soon.

The latest api version [was released and contains the necessary endpoint](https://github.com/dasch-swiss/dsp-api/releases/tag/v30.20.0), are you able switch to it and do not need this additional setup at all?

Reply:
I have thought about that, but I want to mimic the experience we get when we actually call the command. This is not the case if we use latest.